### DISCUSSION TITLE
[Feature]  support alarm for the peak of  certain metric value

### DISCUSSION THREAD

**Discussion by woshioosm** [apache/skywalking#11483] [Ideas]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently, only real-time metric are supported, and there is no support for metrics such as the rate of change (such as a sudden increase in cpm or response time). Can corresponding support be provided
(目前只支持实时的指标，不支持变化率这种指标如 tps 激增，响应时间激增等情况 能否提供相应的支持)

for example:  
The cpm of previous minutes is 10000, but in the following period， the cpm suddenly increased to three(a threshold value) times like 30000， we need to get noticed of this sudden change。

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Could you explain more? If it is a peak in several minutes , it is possible to do through the latest expression based engine(MQE).
Do you check the latest release?

**Reply by woshioosm** [reply]
I also viewed the latest MQE and found that the current dimension also performs various operations in its own time bucket, and cannot be compared with other adjacent time buckets.For example, if the number of requests in the first minute is 100, and suddenly becomes 1000 in the second minute, I believe that this 10-fold surge event also needs to be alerted.I am not sure how to configure this scenario at present.  In short, I hope that we can not only focus on values, but also on the changes in values.  How can we achieve this?
（我也看了下最新的 MQE  ，发现现在的维度也是在自己的时间桶里面进行各种运算，无法和其他相邻时间桶进行比较。  比如说第一分钟的请求数为100， 第二分钟突然变为1000，则我认为这个10倍的激增事件我也需要告警出来。 这种场景目前我不知道如何进行配置。   总之，希望不仅仅关注值，也能关注值的变化，请问如何实现呢？）

**Reply by wu-sheng** [reply]
This is like a query stage range.

@wankai123 We could discuss this in person at Shanghai(SkyWalking summit)

@woshioosm I don't know whether you will join. But this is new, so, hard to say whether and/or how this could be done.

**Reply by woshioosm** [reply]
Thanks for your reply.   I hope that ’Alarm function' can expose all the metrics within the period and allow access through indexes, such as period=n.  This allows users to retrieve the CPM values in each time bucket by using service_cpm.get(0 ~ n-1), and then apply more rich and flexible expressions to these data.
(多谢您的回复， 希望alarm这里可以把 period内的所有指标都暴露出来，并可通过索引访问，如period=n， 可以通过 service_cpm.get(0 ~ n-1) 的方式获取每个time bucket中的cpm值， 之后用户可以对这些数据应用更加丰富和灵活的表达式。)     

I have also registered for this summit and hope to have a live consultation.
（我也报名参加了这次峰会，希望可以现场请教。）

**Reply by wu-sheng** [reply]
I think the last one doesn't make sense. 
The thing you mentioned indicates actually you don't know how MQE and alerting window works.

Let's focus on use case only. If you want to talk about details, you should read codes first, to keep us in the same context.

**Comment by wu-sheng** [comment]
The conclusion is, we could provide rate, increase, and abs functions for MQE to measure the change and/or change rate.

cc @wankai123

**Comment by wu-sheng** [comment]
The feature progress is being trackes by https://github.com/apache/skywalking/issues/11500

**Comment by wu-sheng** [comment]
PR is raised, https://github.com/apache/skywalking/pull/11591, you should have this in the 9.7 release.
